### DISCUSSION TITLE
Provide token based authentication for user administration

### DISCUSSION THREAD

**Discussion by muellerst-hg** [hoppscotch/hoppscotch#3571] [Ideas]
As a HoppScotch administrator
I want to delete users through an API using tokens/credentials
So that I can automate user management

Background:
We have a self-hosted HoppScotch Backend, login exclusively with Microsoft Auth Provider.
The number of users and teams is too large to be handled with the admin dashboard.
Whenever a user has left our organisation, the related HoppScotch user should be deleted.

A solution would be to send queries for user deletion to the Backend GraphQL API and authenticate with a token.

Can you provide any other solution or approach?

**Comment by liyasthomas** [comment]
We have not yet exposed our backend server's API for operations, hence it's unavailable for general use. Providing Hoppscotch's APIs with personal access tokens to manage users and teams is something we can implement in the enterprise edition if there are enough requests.

If you think it would be a great feature for enterprise users feel free to [open a feature request](https://github.com/hoppscotch/hoppscotch/issues/new/choose) with a detailed description of the use case and how it would be beneficial for companies.
